Source/GUI/preferences.cpp

```diff
@@ -175,3 +218,76 @@ void Preferences::OnRejected()
     Load();
 }
 
+void Preferences::on_testConnection_pushButton_clicked()
+{
+    struct UI {
+        static void setSuccess(QLabel* label, QPushButton* button) {
+            label->setStyleSheet("color: green");
+            label->setText("Success!");
+            button->setEnabled(true);
+        }
+        static void setError(QLabel* label, const QString& error, QPushButton* button) {
+            label->setStyleSheet("color: red");
+            label->setText(error);
+            button->setEnabled(true);
+        }
+        static void setChecking(QLabel* label, QPushButton* button) {
+            label->setStyleSheet("");
+            label->setText("Checking..");
+            button->setEnabled(false);
+        }
+    };
+
+    UI::setChecking(ui->connectionTest_label, ui->testConnection_pushButton);
+
+    QNetworkAccessManager manager;
+    QNetworkRequest request(ui->signalServerUrl_lineEdit->text() + "/fileuploads/upload/test");
+    request.setRawHeader("Authorization", "Basic " + QByteArray(QString("%1:%2")
+							 .arg(ui->signalServerLogin_lineEdit->text())
+							 .arg(ui->signalServerPassword_lineEdit->text()).toLocal8Bit().toBase64()));
```

Comment:
As far as I understand, the password is sent in clear text ([Authorization: Basic](https://en.wikipedia.org/wiki/Basic_access_authentication#Client_side)), possibly without secure connection (I see no restriction to HTTPS in signalServerUrl), which is considered as a bad practice nowadays.
The login should be limited to HTTPS or using an algorithm not transmitting the password in clear text and with a nonce ([example](https://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html)).

Note: the QCTools server access from anyone not authorized on the same network (between the client and the server) is maybe not a big issue, but there are plenty of infected machines which scan the network and users usually use the same password on other websites, including more sensitive (sometimes personal) content.